src/analysis/analyzer_gen.ts

```diff
@@ -8,7 +8,7 @@ import * as as from "./analysis_server_types";
 import * as a from "./analyzer";
 
 export abstract class AnalyzerGen {
-	protected abstract notify<T>(subscriptions: ((notification: T) => void)[], params: T);
+	protected abstract notify<T>(subscriptions: ((notification: T) => void)[], params: T): void;
```

Comment:
heh, yep, I commented about that just after. It's a dart script which takes the [API Doc](https://htmlpreview.github.io/?https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/doc/api.html) and generates all the TS types from it. I don't know if there's enough info for us to do this totally accurately (eg. know what can/can't be null) but certainly changes like this are simple :-)